packages/table/src/headers/header.tsx

```diff
@@ -239,8 +239,11 @@ export class Header extends React.Component<IInternalHeaderProps, IHeaderState>
         this.state = { hasValidSelection: this.isSelectedRegionsControlledAndNonEmpty(props) };
     }
 
-    public componentWillReceiveProps(nextProps?: IInternalHeaderProps) {
-        this.setState({ hasValidSelection: this.isSelectedRegionsControlledAndNonEmpty(nextProps) });
+    public componentDidUpdate(_: IInternalHeaderProps, prevState: IHeaderState) {
+        const nextHasValidSection = this.isSelectedRegionsControlledAndNonEmpty(this.props);
+        if (prevState.hasValidSelection !== nextHasValidSection) {
+            this.setState({ hasValidSelection: nextHasValidSection });
```

Comment:
you seem to have reverted the commit which removed the change check. can you explain why please?

Reply:
see https://github.com/palantir/blueprint/pull/3702#discussion_r326727681
removing the checks changes behavior (tests break)